test/property.test.js

```diff
@@ -83,7 +152,103 @@ describe('loopback:property generator', function() {
     });
   });
   
-  it('creates a defaultFn: "now" on date fields if specified', function(done) {
+  it('creates string item typed array', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'options',
+      type: 'array',
+      itemType: 'string',
+      required: 'true',
+      defaultValue: 'AWD,3.2L, navigation'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('options');
+      expect(props.options).to.eql({
+        type: 'array',
+        itemType: 'string',
+        required: true,
+        default: ['AWD','3.2L','navigation']
+      });
+      done();
+    });
+  });
+
+  it('creates number item typed array', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'parts',
+      type: 'array',
+      itemType: 'number',
+      required: 'true',
+      defaultValue: '123456, 98765'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('parts');
+      expect(props.parts).to.eql({
+        type: 'array',
+        itemType: 'number',
```

Comment:
I won't for now, but this string/number typed array syntax stunt is just plainly counter productive and a pain in the behind for all intents and purposes... it seems like a great compact syntax, until you try to do anything useful with the value...